src/features/user/components/ChangeAvatar/ChangeAvatar.tsx

```diff
@@ -0,0 +1,133 @@
+import styled, { css } from "styled-components"
+import ChangeAvatarModal from "../ChangeAvatarModal/ChangeAvatarModal"
+import { removeUserPhoto } from "../../userThunk"
+import { Box, Divider, IconButton, Menu, Tooltip } from "@mui/material"
+import EditRoundedIcon from "@mui/icons-material/EditRounded"
+import DeleteForeverRoundedIcon from "@mui/icons-material/DeleteForeverRounded"
+import PermMediaIcon from "@mui/icons-material/PermMedia"
+import { useTranslation } from "react-i18next"
+import firstLetterToUppercase from "@/utils/firstLetterToUppercase"
+
+const ChangeAvatar = () => {
+  const { t } = useTranslation(["settings"])
+  const dispatch = useAppDispatch()
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false)
+  const {
+    userData: { photo },
+  } = useAppSelector((store) => store.user)
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true)
+  }
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+  const open = Boolean(anchorEl)
+  const handleClickMenu = (event: MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleCloseMenu = () => {
+    setAnchorEl(null)
+  }
+
+  return (
+    <>
+      <Box display="flex" justifyContent={{ xs: "center", sm: "center" }}>
+        <Box position="relative" display="inline-block">
+          <StyledAvatar $size="20rem" src={photo} />
+          <Tooltip
+            title={firstLetterToUppercase(t("common:edit"))}
+            arrow
+            followCursor
+          >
+            <StyledIconButton
+              id="edit-avatar-button"
+              aria-controls={open ? "edit-avatar-menu" : undefined}
+              aria-haspopup="true"
+              aria-expanded={open ? "true" : undefined}
+              onClick={handleClickMenu}
+            >
+              <EditRoundedIcon fontSize="large" color="primary" />
+            </StyledIconButton>
+          </Tooltip>
+          <StyledMenu
+            id="edit-avatar-menu"
+            anchorEl={anchorEl}
+            open={open}
+            onClose={handleCloseMenu}
+            MenuListProps={{
+              "aria-labelledby": "edit-avatar-button",
+              disablePadding: true,
+            }}
+          >
+            <li onClick={handleCloseMenu}>
+              <Button
+                onClick={handleOpenModal}
+                variant="text"
+                size="large"
+                fullWidth
+                startIcon={<PermMediaIcon />}
+              >
+                {t("common:change")}
+              </Button>
+            </li>
+            <Divider />
+            <li onClick={handleCloseMenu}>
+              <Button
+                disabled={!photo}
+                variant="text"
+                size="large"
+                color="error"
+                fullWidth
+                startIcon={<DeleteForeverRoundedIcon />}
+                onAsyncClick={() => dispatch(removeUserPhoto())}
```

Comment:
If youre struggling with passing a dispatch as an async function via props move it to the async function like

const deleteUserPhoto = async()=> { await dispatch(removeUserPhoto))}